proxy/http/HttpSM.h

```diff
@@ -659,4 +684,41 @@ HttpSM::is_transparent_passthrough_allowed()
           ua_session->get_transact_count() == 1);
 }
 
+inline int64_t
+HttpSM::postbuf_read_avail()
+{
+  return this->_postbuf.ua_buffer_reader->read_avail();
+}
```

Comment:
Could we rename this method to something like `postbuf_reader_avail`?

Referencing a debug log message in HttpTunnel.cc:

```
1163 Debug("http_redirect", "[HttpTunnel::producer_handler] post exceeds buffer limit, buffer_avail=%" PRId64
1164                              " reader_avail=%" PRId64 " limit=%" PRId64 "",
1165             sm->postbuf_read_ndone(), sm->postbuf_read_avail(), HttpConfig::m_master.post_copy_size);
```

Here the message uses `"buffer_avail"` and `"reader_avail"`, and it seems clearer.

Reply:
good point !